org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java

```diff
@@ -157,7 +158,7 @@ protected ResourceSet initialValue() {
 					return resourceSet;
 				}
 			};
-			this.executor = Executors.newFixedThreadPool(nThreads);
+			this.executor = Executors.newFixedThreadPool(nThreads, new ThreadFactoryBuilder().setNameFormat("ParallelLoad-%d").build());
```

Comment:
am not sure if we can easily find out if this is cross linking or indexing loader here.

Reply:
Slightly unrelated: Should we reuse the executor for all subsequent builds? Should this be made available from the outside?

Reply:
> find out if this linking or indexing

I don't think that's possible. We'd need an additional key that is passed from the outside.

Reply:
> Should this be made available from the outside?
thats something to think about. but this will lead to even more permantent threads.

Reply:
(delegatingcontentassistcontextfactory already eats up numberoflangsx3